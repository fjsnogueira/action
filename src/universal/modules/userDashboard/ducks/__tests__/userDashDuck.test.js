import reducer, {
  selectNewActionTeam,
  filterTeam,
} from '../userDashDuck';

test('initial state', () => {
  const initialState = reducer();
  expect(initialState).toMatchSnapshot();
});

test('selectNewActionTeam on', () => {
  expect(reducer(undefined, selectNewActionTeam(true))).toMatchSnapshot();
});

test('selectNewActionTeam off', () => {
  const state = reducer(undefined, selectNewActionTeam(true));
  const nextState = reducer(state, selectNewActionTeam(false));
  expect(nextState).toMatchSnapshot();
});

test('filter team on', () => {
  const state = reducer(undefined, filterTeam('123', 'Team A'));
  expect(state).toMatchSnapshot();
});

test('filter team off', () => {
  const state = reducer(undefined, filterTeam('123', 'Team A'));
  const nextState = reducer(state, filterTeam());
  expect(nextState).toMatchSnapshot();
});

test('filter team off bad input', () => {
  const state = reducer(undefined, filterTeam('123', 'Team A'));
  const nextState = reducer(state, filterTeam(null, 'Foo'));
  expect(nextState).toMatchSnapshot();
});
