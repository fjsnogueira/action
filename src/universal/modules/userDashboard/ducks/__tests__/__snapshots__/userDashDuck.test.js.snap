// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter team off 1`] = `
Object {
  "selectingNewActionTeam": false,
  "teamFilterId": null,
  "teamFilterName": "All teams",
}
`;

exports[`filter team off bad input 1`] = `
Object {
  "selectingNewActionTeam": false,
  "teamFilterId": null,
  "teamFilterName": "All teams",
}
`;

exports[`filter team on 1`] = `
Object {
  "selectingNewActionTeam": false,
  "teamFilterId": "123",
  "teamFilterName": "Team A",
}
`;

exports[`initial state 1`] = `
Object {
  "selectingNewActionTeam": false,
  "teamFilterId": null,
  "teamFilterName": "All teams",
}
`;

exports[`selectNewActionTeam off 1`] = `
Object {
  "selectingNewActionTeam": false,
  "teamFilterId": null,
  "teamFilterName": "All teams",
}
`;

exports[`selectNewActionTeam on 1`] = `
Object {
  "selectingNewActionTeam": true,
  "teamFilterId": null,
  "teamFilterName": "All teams",
}
`;
